### PR TITLE
Rbenv fix

### LINKED ISSUE #23: rbenv installs latest ruby, but gem_package installs to system ruby
Maybe I'm missing something in my ansible playbook setup, but what I'm seeing is the following behavior:
1. rbenv role runs, installs ruby 2.1.1
2. gem_package roles run, but rbenv hasn't been evaluated by ansible so the ruby of current install scope is not the new 2.1.1, but the OSX standard /usr/bin/ruby
3. gems are installed, but in OSX system ruby, **not** in the rbenv'd 2.1.1
4. Then I need to manually re-install the gems in a shell that's been launched with rbenv evaluated.

Playbook as follows:

```

---
- hosts: all
  - role: rbenv
  - role: gem_package
    package_name: bundler
```

### PR DESCRIPTION
**Pull request by rjacoby**
Fixes #23.

Ansible's built in `gem` command doesn't run in the context of `rbenv`.

This task wraps the `gem` command in a new login shell to make sure the shims are loaded into path. I made this a separate role in case maintainers don't want to force everyone into using rbenv.

See: https://stackoverflow.com/questions/22115936/install-bundler-gem-using-ansible/22121802#22121802

### CODE REVIEW

**Comment by ninjabong** [comment]
you may wanna reuse the .bash_profile / .zshenv / .profile logic from homebrew_cask/task/main.yml (the one I missed the quote in) as it screws up .profile bash users which is why I made that patch.

the workaround does seem like a good way to deal with the similar problem with homebrew_cask's missing env variable on a clean system/first run resulting in things installing to ~/Applications instead of /Applications.

**Comment by rricard** [comment]
yep @ninjabong is right ! but there's not much to do there for you: just replace one line and it's done !

**Comment by rricard** [comment]
anyway, it's a great improvment but, be sure to run your script in an indepotent way. for that, you can check if a file exists when the command has effectively ran with `creates=<filename>`

**Comment by rjacoby** [comment]
I added the `-role: rbenv` dependency.

As far as idempotency, the `gem` command handles that itself.

In order for us to add a `creates` clause we'd have to find the path to current rbenv gems (not too hard, but `Gem.path` is an array that has several entries, so we pattern match) but also know what version was getting installed (harder).

It would be something like: 

```
creates=`ruby -r rubygems -e "p Gem.path.find{|e| e.match(/\.rbenv/)} + 'gems/{{ package_name }}-{{ SOME WAY OF GETTING GEM VERSION }}'"`
```

and that seems more complex/fragile than just letting `gem install` handle the idempotency (which IMHO it does fine).

File: roles/rbenv_gem_package/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+# Installs a gem into the correct rbenv'd ruby by using a login shell around the command. See:
+#   https://stackoverflow.com/questions/22115936/install-bundler-gem-using-ansible/22121802#22121802
+- name: install {{ package_name }}
+  command: zsh -lc "gem install {{ package_name }}"
```
Comment (rricard) [inline roles/rbenv_gem_package/tasks/main.yml]: dependency upon zsh: we don't want that

Comment (rjacoby) [inline roles/rbenv_gem_package/tasks/main.yml]: Changed to bash, altho OSX ships with both and the rbenv task ensures that dotfiles exist for both.

**Comment by rricard** [comment]
Great ! Seems good to merge

**Pull request merged**